WIP: Bluelink multi region + OAuth2

RustyDust opened the pull request:
## Purpose
Support Hyundai/ Kia bevicles outside EU

## ToDo
1) port login flow from [Hyundai Kia Connect API](https://github.com/Hyundai-Kia-Connect/hyundai_kia_connect_api) for
    - ~~Europe~~ ✅
    - USA
    - Canada
    - Australia
    - New Zealand
    - China
    - India
2) During or after 1., refactor to use `OAuth2.Config` instead of legacy config, depending on
    - how much the region specific implementations differ from each other and 
    - how well they fit/work with the existing OAuth2 code

RustyDust committed:
First draft of implementation direction

Signed-off-by: Stefan Rubner <stefan@whocares.de>

RustyDust committed:
Not all regions support all brands :/

Signed-off-by: Stefan Rubner <stefan@whocares.de>

RustyDust commented:
@andig Grobes Outline, wie ich mir das vorstellen würde, damit du schon frühzeitig sagen kannst "so nicht" ;)

andig commented on vehicle/bluelink/params.go:
```diff
@@ -4,3 +4,18 @@ const (
 	HyundaiAppID = "014d2225-8495-4735-812d-2616334fd15d"
 )
+
+const (
+	BlueLinkRegionEurope     = "Europe"      // OAuth2
+	BlueLinkRegionCanada     = "Canada"      // legacy
+	BlueLinkRegionUSA        = "USA"         // legacy + different method for Kia / Hyundai
+	BlueLinkRegionChina      = "China"       // OAuth2
+	BlueLinkRegionAustralia  = "Australia"   // OAuth2
+	BlueLinkRegionIndia      = "India"       // OAuth2
+	BlueLinkRegionNewZealand = "New Zealand" // legacy + no Hyundai
+)
```
Brauchst du die? Im Zweifel kommt die Region ja aus Dropdown im Template

RustyDust replied:
Ok, dann in zwei Schritten, finde ich gut.

RustyDust committed:
- add region to templates
- shorten some constant's names
- rename brandLogin to brandLoginEU
- add region translations to de.json/en.json (conflicts Weblate?)
- low hanging fruit: support brand "Genesis Motor"

RustyDust committed:
- add region to templates
    - shorten some constant's names
    - rename brandLogin to brandLoginEU
    - add region translations to de.json/en.json (conflicts Weblate?)
    - low hanging fruit: support brand "Genesis Motor"

Signed-off-by: Stefan Rubner <stefan@whocares.de>

RustyDust committed:
- move login functionality for EU to `LoginEU(...)`
- create initial `LoginCA()`
- minor cleanups

Signed-off-by: Stefan Rubner <stefan@whocares.de>

RustyDust committed:
Merge branch 'evcc-io:master' into kia_hyundai_oauth2

RustyDust committed:
Incorporate hyundai fixes from master

Signed-off-by: Stefan Rubner <stefan@whocares.de>

RustyDust committed:
Merge branch 'kia_hyundai_oauth2' of github.com:RustyDust/evcc into kia_hyundai_oauth2

RustyDust committed:
First go at AU bluelink login for Hyundai / Kia

Signed-off-by: Stefan Rubner <stefan@whocares.de>

RustyDust committed:
Merge branch 'evcc-io:master' into kia_hyundai_oauth2
